tests/reference/asr-structs_02-f95782c.stderr

```diff
@@ -1,5 +1,5 @@
-semantic error: struct S s must be initialized a value
- --> tests/errors/structs_02.py:8:5
+semantic error: S not supported yet in Attribute.
+ --> tests/errors/structs_02.py:9:5
   |
-8 |     s: S
-  |     ^^^^ 
+9 |     s.x = 2
```

Comment:
Good test --- when `S` is not `lpython.S`, but rather a user defined struct. We need to distinguish between the two cases.

Reply:
Just to clarify , does this mean we should always have the import statement `from lpython import S` in any program we write ? Should using the `S` annotation without the import statement work ? 

Cause in python when we define a class with the same name as a built-in primitive data type, we essentially override the reference to that primitive type with our class definition.

This error essentially spawns from the following block in `get_type_from_var_annotation` . 
```
        } else if (var_annotation == "S") {
            type = ASRUtils::TYPE(ASR::make_SymbolicExpression_t(al, loc));
```

Here we could add a check if the import statement is present or not !

Reply:
> Should using the S annotation without the import statement work ?

Yup, I think it might work, the import statement is used for CPython to compile.

Reply:
Yup, that is one way, you can also do:
```diff
diff --git a/src/lpython/semantics/python_ast_to_asr.cpp b/src/lpython/semantics/python_ast_to_asr.cpp
index 849c157d3..c7a4b778d 100644
--- a/src/lpython/semantics/python_ast_to_asr.cpp
+++ b/src/lpython/semantics/python_ast_to_asr.cpp
@@ -1040,8 +1040,6 @@ public:
             type = ASRUtils::make_Array_t_util(al, loc, type, dims.p, dims.size());
         } else if (var_annotation == "CPtr") {
             type = ASRUtils::TYPE(ASR::make_CPtr_t(al, loc));
-        } else if (var_annotation == "S") {
-            type = ASRUtils::TYPE(ASR::make_SymbolicExpression_t(al, loc));
         } else if (var_annotation == "pointer") {
             LCOMPILERS_ASSERT(n_args == 1);
             AST::expr_t* underlying_type = m_args[0];
@@ -1073,6 +1071,8 @@ public:
                         }
                     }
                 }
+            } else if (var_annotation == "S") {
+                type = ASRUtils::TYPE(ASR::make_SymbolicExpression_t(al, loc));
             }
             if( raise_error ) {
                 throw SemanticError("Unsupported type annotation: " + var_annotation, loc);
```

Reply:
It checks whether the S struct is defined, if not then assume it as a symbolic type.

Reply:
Yes , this should do the job !